tests/vmi_sound_test.go

```diff
@@ -40,51 +40,37 @@ import (
 	"kubevirt.io/kubevirt/tests/testsuite"
 )
 
-var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-compute] Sound", decorators.SigCompute, func() {
-
-	var virtClient kubecli.KubevirtClient
-
-	BeforeEach(func() {
-		virtClient = kubevirt.Client()
-	})
+var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-compute] Sound", decorators.SigCompute, decorators.Conformance, func() {
```

Comment:
Why is this test so critical to merit `decorators.Conformance` ?
I'm not even sure it is worth starting a fresh VMI for it. Each started VM represents a race condition, wasted time, etcd pressure, CI budget etc, just to test that libvirt does what it has been doing for decades (converting domxml to proper guests).

Reply:
First of all, now I see I need to move the decorator to specific tests otherwise we ask for troubles.

For the reasoning, I believe that this is one of the API behaviors that users will rely on. While Kubevirt doesn't act as VDI solution the sound device is important for users that want to have VDI implemented on top of it. Would you disagree? (I will amend the commit with reasoning based on the discussion)

Reply:
> Would you disagree?

Indeed. Sound is important. I do not suggest to drop this feature from KubeVirt. At this point in time I don't even suggest to drop it from e2e tests (though I reserve the right to do so in the future).

I'm stating that this is a `crit:medium`, niche and ancient feature, where KubeVirt is doing basically nothing. Therefore I don't think that it is worth running a VM in the Conformance suite.

Reply:
I believe the VDI use case does justify to have one test in Conformance suite.

Reply:
I'm very conflicted here actually. Maybe I will include it because we have no other test for VDI use case, and since it is an ancient feature, users expect this to work

Reply:
> users expect this to work

Hence we don't drop it from e2e.

I'd add one last motivation for my opinion: historically, we are very quick to add tests and very slow to delete redundant ones. Heck, we still have tests that take minutes to finish. If you decide to add this test to Conformance in a month's time and give a reason, nobody is likely to object.

Reply:
I have a stronger opinion, I would drop this e2e test as I think ultimately this tests libvirt rather than KubeVirt.

To cover KubeVirt I think a unit test where we check that we produce the right domain xml for libvirt would be enough, I don't see much benefit in having this as an e2e test or even as a conformance test.

Reply:
I think we need to keep a sound functest to ensure the sound card emulator doesn't get removed by qemu / centos / us. Maybe it also needs the right version of some libs to be present on the filesystem and so on.
However, we probably don't need a dedicated test.
No opinion on conformance, but we should definitely only decorate specific tests and not entire `Decribe`s.

Reply:
>I'd add one last motivation for my opinion: historically, we are very quick to add tests and very slow to delete redundant ones. Heck, we still have tests that take minutes to finish. If you decide to add this test to Conformance in a month's time and give a reason, nobody is likely to object.

@dankenigsberg I hope we can change it and I go as far as saying we already do some changes here. Take for example how many test we are removing or going to remove as part of my and @fossedihelm PRs around conformance...

Reply:
>

I have a stronger opinion, I would drop this e2e test as I think ultimately this tests libvirt rather than KubeVirt.

To cover KubeVirt I think a unit test where we check that we produce the right domain xml for libvirt would be enough, I don't see much benefit in having this as an e2e test or even as a conformance test.

@acardace This is not enough, we can't blindly trust everyone and our self that we don't drop something through all the components or as @jean-edouard points out that someone else will not drop dependency or such.
@jean-edouard +1 for your answer

The goal which I think we should strive for is to have minimum of e2e tests and if there are variant of a setting we should cover it by unit tests. Variants should be included only if they differ too much,...  Ideally we would have light-weight integration tests with libvirt but who has time to pick this up now?

Reply:
Yes, you are making great progress on this goal of reducing redundant tests, making tests quicker and more reliable. (Still, I think that it is easier for a developer to add another VM startup on a test lane than to remove it from the lane.)

Reply:
Note sure we are steering towards one or other opinion. @vladikr maybe you can be the deciding vote?

Reply:
> @acardace This is not enough, we can't blindly trust everyone and our self that we don't drop something through all the components or as @jean-edouard points out that someone else will not drop dependency or such. @jean-edouard +1 for your answer

Even if I think we should kinda trust libvirt blindly as it's such a hard dependency for us (we even rely on a fixed version of libvirt to exists for a specific KubeVirt release) that's not my point entirely, I think this is more of a fit as an integration test rather than an e2e test but I do know we don't have integration tests right now.

Reply:
I initially (mistakenly) thought that the sound card is needed for PCI passthought testing - where we use the sound card as a dummy pci device. 
From my pov, we don't need to test the sound in conformance and only focus on testing functionalities that are in KubeVirt's hands to break - such as networking, migrations, PCI passthought (@xpivarc said that we can't)
I would perhaps configure the sound card as part of another existing tests, instead - if this is really needed.

Reply:
Done, ptal